app/src/main/java/org/dslul/openboard/inputmethod/keyboard/KeyboardView.java

```diff
@@ -517,23 +534,37 @@ protected void onDrawKeyTopVisuals(@NonNull final Key key, @NonNull final Canvas
         }
     }
 
-    // Draw popup hint "..." at the bottom right corner of the key.
     protected void drawKeyPopupHint(@NonNull final Key key, @NonNull final Canvas canvas,
             @NonNull final Paint paint, @NonNull final KeyDrawParams params) {
         if (TextUtils.isEmpty(mKeyPopupHintLetter)) {
             return;
         }
+        final SharedPreferences prefs = DeviceProtectedUtils.getSharedPreferences(getContext());
+        final String style = prefs.getString(Settings.PREF_THEME_STYLE, STYLE_MATERIAL);
```

Comment:
Getting and reading shared preferences potentially on every call to `onDrawKey` seems rather inefficient. Consider that `onDrawKey` is called for every key when the keyboard is redrawn, and the keyboard is redrawn on every key press and on every key release.
Better use the cached style in `mColors`.